blockchain/blockchain.go

```diff
@@ -1397,6 +1415,41 @@ func (bc *BlockChain) gcCachedNodeLoop() {
 	}()
 }
 
+func (bc *BlockChain) pruneTrieNodeLoop() {
+	// ReadPruningMarks(1, limit) is very slow because it iterates over the most of MiscDB.
```

Comment:
Is this an acceptable speed? Don't we need to apply more fast retrieval methods such as binary search between [1, latest]?

Reply:
Acceptable to me. Before having `startNum` it took around 3 seconds per loop and most of the time was spent in ReadPruningMarks. But now it only happens at the early stage of the program.

Reply:
Update: Tested with a node with an existing database. The first pruning loop right after the program started took 1m17s. After that, it takes around 50ms every 128 blocks.

Considering that the pruning loop runs in the background, I think 1 minute is tolerable.
```
INFO[07/25,06:45:00 Z] [5] Pruned trie nodes                         number=49000064 start=1 limit=48827264 count=12795 elapsed=1m17.616352458s
INFO[07/25,06:45:00 Z] [5] Pruned trie nodes                         number=49000192 start=48827265 limit=48827392 count=17332 elapsed=55.195561ms
INFO[07/25,06:45:00 Z] [5] Pruned trie nodes                         number=49000320 start=48827393 limit=48827520 count=15679 elapsed=63.148902ms
INFO[07/25,06:45:00 Z] [5] Pruned trie nodes                         number=49000448 start=48827521 limit=48827648 count=17534 elapsed=46.980492ms
```

Reply:
@blukat29 Thanks for the update. 1 min at the first booting sounds acceptable to me as well. For a better understanding, be aware that heavy disk read will be conducted at that time.